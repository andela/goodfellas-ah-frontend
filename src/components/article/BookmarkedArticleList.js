import React, { Fragment } from 'react';
import ProfileArticle from './Profile';
import icons from '../../assets/icons.svg';

const BookmarkedArticleList = (props) => {
  const {
    articles,
  } = props;
  return (
    <Fragment>
      {
        articles.length === 0
          ? (
            <div className="no-record centralizer">
              <svg className="icon">
                <use xlinkHref={`${icons}#sad`} />
              </svg>&nbsp;&nbsp;
              <span>You have not bookmarked any articles</span>
            </div>
          )
          : articles.map((article) => {
            const { article: { user: { firstname, lastname, profile: { image } } } } = article;
            return (<ProfileArticle key={article.id} author={`${firstname} ${lastname}`} authorImage={image} type="favoriteArticle" article={article.article} />)
          })
      }
    </Fragment>);
};

export default BookmarkedArticleList;
