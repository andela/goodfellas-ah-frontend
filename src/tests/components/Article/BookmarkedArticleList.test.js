import React from 'react';
import { shallow } from 'enzyme';
// eslint-disable-next-line import/no-extraneous-dependencies
import toJson from 'enzyme-to-json';
import BookmarkedArticleList from '../../../components/article/BookmarkedArticleList';
import * as profileData from '../../mock/profileData';

describe('BookmarkedArticleList UI', () => {
  const emptyListComponent = shallow(<BookmarkedArticleList userId={1} articles={[]} article={{ authorId: 1, slug: 'some-sug', bookmarked: [] }} />);
  const listComponent = shallow(<BookmarkedArticleList userId={1} articles={profileData.bookmarkedArticles} article={{ authorId: 1, slug: 'some-sug', bookmarked: [] }} />);
  it('view should render as expected', () => {
    const tree = toJson(emptyListComponent);
    expect(tree).toMatchSnapshot();
  });
  it('view should render as expected', () => {
    const tree = toJson(listComponent);
    expect(tree).toMatchSnapshot();
  });
});
