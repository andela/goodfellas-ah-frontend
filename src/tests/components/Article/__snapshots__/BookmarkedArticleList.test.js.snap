// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkedArticleList UI view should render as expected 1`] = `
<Fragment>
  <div
    className="no-record centralizer"
  >
    <svg
      className="icon"
    >
      <use
        xlinkHref="icons.svg#sad"
      />
    </svg>
      
    <span>
      You have not bookmarked any articles
    </span>
  </div>
</Fragment>
`;

exports[`BookmarkedArticleList UI view should render as expected 2`] = `
<Fragment>
  <withRouter(Profile)
    article={
      Object {
        "authorId": 1,
        "averageRating": null,
        "body": "This is the fine long body this is the fine long body  this is the fine long body  this is the fine long body  this is the fine long body ",
        "comments": Array [],
        "createdAt": "2018-11-09T15:08:00.560Z",
        "description": " this is the fine long body  this is the fine long body ",
        "favorited": null,
        "favoritesCount": null,
        "id": 2,
        "image": null,
        "reactions": Array [],
        "read_time": "1 minute",
        "slug": "this-is-a-fine-title-which-i-would-really-love-to-have",
        "tagList": null,
        "title": "This is a fine title which I would really love to have",
        "updatedAt": "2018-11-09T15:08:00.560Z",
        "user": Object {
          "email": "damafeez11@gmail.com",
          "firstname": "Afeez1",
          "lastname": "Awoyemi",
          "notificationSettings": Array [
            "email",
            "inApp",
          ],
          "profile": Object {
            "image": "image",
          },
          "role": "User",
        },
      }
    }
    author="Afeez1 Awoyemi"
    authorImage="image"
    type="favoriteArticle"
  />
</Fragment>
`;
